### PR TITLE
OS-8502 Add support for ED25519 keys, remove unused key checking code

### PR DESCRIPTION
**Pull request by travispaul**
Adds support for ED25519 keys and removes per-key-type related code that is no longer used.

This PR is related to TritonDataCenter/illumos-joyent/pull/462, there will be additional PRs (cloudapi, ufds, etc) to enable ED25519 support throughout the Triton stack.

See [OS-8502](https://mnx.atlassian.net/browse/OS-8502?focusedCommentId=123016) for testing notes.

[OS-8502]: https://mnx.atlassian.net/browse/OS-8502?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: openssh/Patches/0019-PubKeyPlugin-support.patch [hunk 1]
```diff
@@ -201,35 +138,35 @@ index 5d59febc..18a93889 100644
...
+@@ -565,7 +566,7 @@ typedef enum {
  	sAllowStreamLocalForwarding, sFingerprintHash, sDisableForwarding,
  	sExposeAuthInfo, sRDomain, sPubkeyAuthOptions, sSecurityKeyProvider,
  	sRequiredRSASize, sChannelTimeout, sUnusedConnectionTimeout,
 -	sDeprecated, sIgnore, sUnsupported
 +	sPubKeyPlugin, sDeprecated, sIgnore, sUnsupported
  } ServerOpCodes;
  
-@@ -766,6 +767,7 @@ static struct {
+ #define SSHCFG_GLOBAL		0x01	/* allowed in main section of config */
```
Comment (danmcd) [inline openssh/Patches/0019-PubKeyPlugin-support.patch:160]: Why is this here?

**Review by danmcd — COMMENTED** [review]
Only one question, plus testing notes update requested per TritonDataCenter/illumos-joyent#462

**Review by bahamat — CHANGES_REQUESTED** [review]

File: openssh/Patches/0019-PubKeyPlugin-support.patch [hunk 2]
```diff
@@ -119,76 +103,29 @@ index 5d59febc..18a93889 100644
...
 +		dlclose(handle);
-+		ecdsa_sym = NULL;
-+		rsa_sym = NULL;
 +		univ_sym = NULL;
 +		handle = NULL;
 +	}
 +
 +	if (success) {
-+		verbose("Found matching %s key: %s", sshkey_type(key), fp);
++		verbose("Found matching %s key", sshkey_type(key));
```
[the remark this answers is not in the document]
Comment (travispaul) [reply openssh/Patches/0019-PubKeyPlugin-support.patch:118]: That had been printing `(null)` since the `sshd_user_key_allowed` was introduced and started being used. The `fp` pointer was no longer being set.

```
Found matching RSA key: (null)
```

I actually went back and captured logs from before the changes to confirm as I first thought it was odd.

File: openssh/Patches/0019-PubKeyPlugin-support.patch (hunk 1 above)
Comment (travispaul) [reply openssh/Patches/0019-PubKeyPlugin-support.patch:160]: I think this could be the difference in git versions. I didn't alter `serverconf.c` but I did regenerate the changes by applying the original with gpatch, making my changes, then created this file via `git format-patch`.

Let me know if you'd prefer me to just take everything for `serverconf.c` or `servconf.h` from the original patch. I've not modified anything in those files so I can manually edit the patch and retest.

Comment (danmcd) [reply openssh/Patches/0019-PubKeyPlugin-support.patch:160]: I was expecting there to be no *additions*.  If it builds I guess(?) It's okay, but I can imagine maybe gettings a double-define error in compilation.

File: openssh/Patches/0019-PubKeyPlugin-support.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (bahamat) [reply openssh/Patches/0019-PubKeyPlugin-support.patch:118]: Gotcha. All good then.

File: openssh/Patches/0019-PubKeyPlugin-support.patch (hunk 1 above)
Comment (danmcd) [reply openssh/Patches/0019-PubKeyPlugin-support.patch:160]: DOH!  Upon looking at the raw diff file... I see I was viewing the + sign in the wrong column.

**Review by danmcd — APPROVED** [review]
Ship it!
